src/v2/Components/NavBar/NavItem.tsx

```diff
@@ -67,81 +127,92 @@ export const NavItem: React.FC<NavItemProps> = ({
     }
   }
 
+  const containerRef = useRef<null | HTMLDivElement>(null)
+
+  // Close the subnav if it is open and our focus moves outside of it
+  useEffect(() => {
+    const checkFocus = () => {
+      if (!containerRef?.current) return // Not mounted
+      if (!isVisible) return // Isn't open
+      if (containerRef.current.contains(document.activeElement)) return // Focus is within sub-nav
+
+      setIsVisible(false)
+    }
+
+    document.addEventListener("focus", checkFocus, true)
+    return () => {
+      document.removeEventListener("focus", checkFocus)
+    }
+  }, [isVisible])
+
+  const handleClick = () => {
+    onClick && onClick()
+    !!Menu && setIsVisible(prevIsVisible => !prevIsVisible)
+    trackClick()
+  }
+
+  // Add a very short delay to avoid trapping the mouse and capturing intent
+  const openDelayRef = useRef<null | ReturnType<typeof setTimeout>>(null)
+  // Add a slightly longer delay to improve UX when moving at an angle off HitArea
+  const closeDelayRef = useRef<null | ReturnType<typeof setTimeout>>(null)
+
+  const handleMouseEnter = () => {
+    openDelayRef.current = setTimeout(() => {
+      setIsVisible(true)
+    }, 50)
+    closeDelayRef?.current && clearTimeout(closeDelayRef.current)
```

Comment:
This is just a matter of pref, but I always find these naked guards somewhat hard to read later versus a conventional `if`. (its fine tho)